ortools/linear_solver/xpress_interface.cc

```diff
@@ -72,6 +72,32 @@ int XPRSsetobjoffset(const XPRSprob& mLp, double value) {
   return 0;
 }
 
+void XPRS_CC optimizermsg(XPRSprob prob, void* data, const char* sMsg, int nLen,
+                          int nMsgLvl) {
+  switch (nMsgLvl) {
+      /* Print Optimizer error messages and warnings */
+    case 4: /* error */
+      printf("%*s\n", nLen, sMsg);
```

Comment:
You should use `LOG(INFO) <<`  etc instead of a plain printf

Reply:
@Mizux maybe I misunderstood the idea behind `LOG(...)`, but when I use it instead of `printf()`, the log on the screen becomes "dirty" with a prefix. Example:

```I1117 17:41:14.510831 11784 xpress_interface.cc:93] Starting concurrent solve with dual and primal
I1117 17:41:14.514353 11784 xpress_interface.cc:93]
I1117 17:41:14.514495 11784 xpress_interface.cc:93]               Concurrent-Solve,   0s
I1117 17:41:14.514615 11784 xpress_interface.cc:93]             Dual                      Primal
I1117 17:41:14.514733 11784 xpress_interface.cc:93]     objective   dual inf
I1117 17:41:14.514852 11784 xpress_interface.cc:93]  D  295408.67   .0000000 |
I1117 17:41:14.514969 11784 xpress_interface.cc:93] ------- optimal -------- | ----- interrupted ------
I1117 17:41:14.515080 11784 xpress_interface.cc:93] Concurrent statistics:
I1117 17:41:14.515205 11784 xpress_interface.cc:93]       Dual: 273 simplex iterations, 0.00s
I1117 17:41:14.515852 11784 xpress_interface.cc:93]     Primal: 73 simplex iterations, 0.00s
I1117 17:41:14.517642 11784 xpress_interface.cc:93] Optimal solution found
I1117 17:41:14.517849 11784 xpress_interface.cc:93]
I1117 17:41:14.518322 11784 xpress_interface.cc:93]    Its         Obj Value      S   Ninf  Nneg   Sum Dual Inf  Time
I1117 17:41:14.518896 11784 xpress_interface.cc:93]    273       295408.6693      D      0     0        .000000     0
I1117 17:41:14.519452 11784 xpress_interface.cc:93] Dual solved problem
I1117 17:41:14.519996 11784 xpress_interface.cc:93]   273 simplex iterations in 0s
I1117 17:41:14.520515 11784 xpress_interface.cc:93]
I1117 17:41:14.521243 11784 xpress_interface.cc:93] Final objective                       : 2.954086692965932e+05
I1117 17:41:14.523621 11784 xpress_interface.cc:93]   Max primal violation      (abs/rel) : 8.882e-16 / 8.882e-16
I1117 17:41:14.526625 11784 xpress_interface.cc:93]   Max dual violation        (abs/rel) :       0.0 /       0.0
I1117 17:41:14.532981 11784 xpress_interface.cc:93]   Max complementarity viol. (abs/rel) :       0.0 /       0.0
I1117 17:41:14.533523 11784 xpress_interface.cc:93] All values within tolerances
I1117 17:41:14.534399 11784 xpress_interface.cc:93]
I1117 17:41:14.534528 11784 xpress_interface.cc:93] Starting root cutting & heuristics
I1117 17:41:14.535095 11784 xpress_interface.cc:93]
I1117 17:41:14.535377 11784 xpress_interface.cc:93]  Its Type    BestSoln    BestBound   Sols    Add    Del     Gap     GInf   Time
I1117 17:41:14.542755 11784 xpress_interface.cc:93] k         318320.9350  295408.6693      1                  7.20%       0      0
I1117 17:41:14.548524 11784 xpress_interface.cc:93] c         295530.9281  295408.6693      2                  0.04%       0      0
I1117 17:41:14.612131 11784 xpress_interface.cc:93] R         295516.7604  295476.4872      3                  0.01%       0      0
I1117 17:41:14.613627 11784 xpress_interface.cc:93] R         295480.7200  295476.4872      4                  0.00%       0      0
I1117 17:41:14.614469 11784 xpress_interface.cc:93] STOPPING - MIPRELSTOP target reached (MIPRELSTOP=0.0001  gap=1.43251e-005).
I1117 17:41:14.628461 11784 xpress_interface.cc:93]  *** Search completed ***     Time:     0 Nodes:          0
I1117 17:41:14.628756 11784 xpress_interface.cc:93] Number of integer feasible solutions found is 4
I1117 17:41:14.629641 11784 xpress_interface.cc:93] Best integer solution found is  295480.7200
I1117 17:41:14.630493 11784 xpress_interface.cc:93] Best bound is  295476.4872
I1117 17:41:14.634817 11784 xpress_interface.cc:93] Uncrunching matrix```

Is there a way to "hide" this prefix and make the log "cleaner"? 

Thanks

Reply:
there is a command line flags --log_prefix=false

Reply:
Now, on glop and CP-SAT, we use a SolverLogger class, that can be disabled, and that will print out to stdout if wanted.

So we use LOG() for error messages, warnings, exceptions, and SolverLogger (see SOLVER_LOG macro) for solver output.

Reply:
you can also try in python (or other binding)
https://github.com/google/or-tools/blob/2be8529ec34885c413aec758846dbffbfc40a4a9/ortools/linear_solver/samples/basic_example.py#L18

https://github.com/google/or-tools/blob/2be8529ec34885c413aec758846dbffbfc40a4a9/ortools/linear_solver/samples/basic_example.py#L66-L70

Reply:
the --log_prefix=false command works just fine, so I added `LOG(INFO)` and `LOG(WARNING)` instead of `printf()` command

Thanks!